### PR TITLE
Circleci project setup

### PR DESCRIPTION
**Pull request by niyongaboaristide17**
# What does this PR do?
CircleCI is required in this project. It helps for continious integration
# Description of Task to be completed?
- [x] circleCI configuration
# How should this be manually tested?
- Click on circleCI badge in the read me
## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Any background context you want to provide?

# What are the relevant pivotal tracker stories?
- https://www.pivotaltracker.com/story/show/181339561
# Screenshots (if appropriate)

Screenshot from 2022-03-01 20-05-21

### CODE REVIEW

**Review by ericmaniraguha — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -1,3 +1,4 @@
+[![CircleCI](https://circleci.com/gh/atlp-rwanda/elites-bn-be.svg?style=svg&circle-token=f7c825d627bd711423c163f838796046bbeb5648)](https://circleci.com/gh/atlp-rwanda/elites-bn-be)
```
Comment (CYIMANA-Faisal) [inline README.md]: This badge should reflect to the circle CI checks

File: package.json [hunk 2]
```diff
@@ -10,18 +10,19 @@
...
     "start": "npm run build && better-npm-run prod",
     "dev": "better-npm-run dev",
-    "test": "better-npm-run test"
+    "test": "better-npm-run test",
+    "coverage": "better-npm-run cov"
   },
   "betterScripts": {
     "dev": {
-      "command": "nodemon  --exec npx babel-node ./src/app.js",
+      "command": "nodemon --exec babel-node ./src/app.js",
```
Comment (CYIMANA-Faisal) [inline package.json]: why removing npx
any reason

File: package.json [hunk 3]
```diff
@@ -10,18 +10,19 @@
...
       "env": {
         "NODE_ENV": "development",
         "PORT": 3000
       }
     },
     "test": {
-      "command": "nyc --reporter=html --reporter=text mocha --requre @babel/register ./tests/* --timeout 1000000 --exit || true",
+      "command": "nyc --reporter=html --reporter=text --reporter=lcov --reporter=text-summary --check-coverage=true mocha --require @babel/register ./tests/* --timeout 1000000 --exit || true",
```
Comment (CYIMANA-Faisal) [inline package.json]: again can you clarify to why this line. just wondering

File: src/controllers/test.controllers.js [hunk 4]
```diff
@@ -0,0 +1,3 @@
+export function testFunc(){
```
Comment (CYIMANA-Faisal) [inline src/controllers/test.controllers.js:1]: all the unit testing should be in testes

File: tests/test.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
+import chai from 'chai';
```
Comment (CYIMANA-Faisal) [inline tests/test.js]: Please you have to have the same formats for the tests

**Review by CYIMANA-Faisal — CHANGES_REQUESTED** [review]
Good work and please work on the reviews

**Review by ericmaniraguha — APPROVED** [review]

**Review by ericmaniraguha — COMMENTED** [review]
It is fine.

**Pull request closed without merging**